### PR TITLE
Improve SAML tests resiliency to auto-formatting

### PR DESCRIPTION
**Pull request by pugnascotia**
The SAML tests have large XML documents within which various parameters
are replaced. At present, if these test are auto-formatted, the XML
documents get strung out over many, many lines, and are basically
illegible.

Fix this by using named placeholders for variables, and indent the
multiline XML documents.

The tests in `SamlSpMetadataBuilderTests` deserve a special mention,
because they include a number of certificates in Base64. I extracted
these into variables, for additional legibility.

### CODE REVIEW

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/saml/SamlSpMetadataBuilderTests.java [hunk 1]
```diff
@@ -306,4 +373,14 @@ public void testAttributeNameIsRequired() {
     private void assertValidXml(String xml) throws Exception {
         SamlUtils.validate(new ByteArrayInputStream(xml.getBytes(StandardCharsets.UTF_8)), SamlMetadataCommand.METADATA_SCHEMA);
     }
+
+    private String joinCertificateLines(String... lines) {
+        return Arrays.stream(lines).collect(Collectors.joining(System.lineSeparator()));
+    }
+
+    private String normaliseXml(String input) {
```
Comment (rjernst) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/saml/SamlSpMetadataBuilderTests.java]: I see you use that "other" English :)

**Review by rjernst — APPROVED** [review]
LGTM

File: test/framework/src/main/java/org/elasticsearch/common/util/NamedFormatter.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            if (match.group(1) != null) {
+                return match.group(1);
+            }
+
+            final String paramName = match.group(3);
+            if (values.containsKey(paramName) == true) {
+                return values.get(paramName).toString();
+            }
+
+            throw new IllegalArgumentException("No parameter value for %(" + paramName + ")");
```
Comment (jkakavas) [inline test/framework/src/main/java/org/elasticsearch/common/util/NamedFormatter.java]: :+1:

**Review by jkakavas — APPROVED** [review]
LGTM, and makes tests much more readable - thanks @pugnascotia

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/saml/SamlSpMetadataBuilderTests.java (hunk 1 above)
Comment (pugnascotia) [reply x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/saml/SamlSpMetadataBuilderTests.java]: You mean the correct one? :trollface:

**Pull request merged**